pkg/asset/manifests/openstack/cluster.go

```diff
@@ -33,35 +34,46 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 			},
 		},
 		Spec: capo.OpenStackClusterSpec{
-			CloudName: cloudName,
-			IdentityRef: &capo.OpenStackIdentityReference{
-				Kind: "Secret",
-				Name: clusterID.InfraID + "-cloud-config",
+			IdentityRef: capo.OpenStackIdentityReference{
+				Name:      clusterID.InfraID + "-cloud-config",
+				CloudName: cloudName,
 			},
 			// We disable management of most networking resources since either
 			// we (the installer) will create them, or the user will have
 			// pre-created them as part of a "Bring Your Own Network (BYON)"
 			// configuration
-			ManagedSecurityGroups:      false,
-			DisableAPIServerFloatingIP: true,
+			ManagedSecurityGroups:      nil,
+			DisableAPIServerFloatingIP: ptr.To(true),
 			// TODO(stephenfin): update when we support dual-stack (there are
 			// potentially *two* IPs here)
-			APIServerFixedIP:  openstackInstallConfig.APIVIPs[0],
-			DNSNameservers:    openstackInstallConfig.ExternalDNS,
-			ExternalNetworkID: openstackInstallConfig.ExternalNetwork,
+			APIServerFixedIP: &openstackInstallConfig.APIVIPs[0],
+			ExternalNetwork:  &capo.NetworkParam{Filter: &capo.NetworkFilter{Name: openstackInstallConfig.ExternalNetwork}},
```

Comment:
There can be scenarios that the externalNetwork is not specified in the install-config, such as usage of Provider networks. We need to only add external network in case it's defined otherwise we would see the following issue, which happened on the dualstack job:
`level=error msg=failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to create infrastructure manifest: OpenStackCluster.infrastructure.cluster.x-k8s.io "lwbzl5lg-0f699-c5xh5" is invalid: spec.externalNetwork.filter: Invalid value: 0: spec.externalNetwork.filter in body should have at least 1 properties`

Reply:
I suspect that there might also something top be done for dualstack in APIServerFixedIP?

Reply:
I think the `APIServerFixedIP` which in the end becomes `ControlPlaneEndpoint.Host` is only actually used in case we would create an API LB in CAPO. I don't think we need to change something related to `APIServerFixedIP`.